core/src/test/java/com/yahoo/imapnio/async/data/StoreResultTest.java

```diff
@@ -0,0 +1,93 @@
+        final StoreResult storeResult = new StoreResult(1L, imapResponses, modifiedMsgNums);
+        final long highestModSeq = storeResult.getHighestModSeq();
+        final List<IMAPResponse> responses = storeResult.getIMAPResponses();
+
+        Assert.assertEquals(responses.size(), 2, "Result mismatched.");
+        Assert.assertEquals(highestModSeq, 1L, "Result mismatched.");
+        Assert.assertTrue(responses.get(0).keyEquals("FETCH"), "Result mismatched.");
+        final long fetchedModSeq = new FetchResponse(responses.get(0)).getItem(MODSEQ.class).modseq;
+        Assert.assertEquals(fetchedModSeq, 12121231000L, "Result mismatched.");
+        Assert.assertTrue(responses.get(1).keyEquals("VANISHED"), "Result mismatched.");
+        final String modifiedMsgsets = MessageNumberSet.buildString(storeResult.getModifiedMsgSets());
+        Assert.assertEquals(storeResult.getModifiedMsgSets().length, 1, "Result mismatched.");
+        Assert.assertEquals(modifiedMsgsets, "1", "Result mismatched.");
+    }
+
+    /**
+     * Tests StoreResult constructor and getters when not passing highest mod sequence, empty fetch responses collection,
+     * and empty modified message number collection.
+     */
+    @Test
+    public void testStoreResultIgnoreHighestModSeq() {
+        final MessageNumberSet[] msgSets = new MessageNumberSet[0];
+        final StoreResult infos = new StoreResult(Collections.emptyList(), msgSets);
+        final MessageNumberSet[] modifiedMsgsets = infos.getModifiedMsgSets();
+        final List<IMAPResponse>responses = infos.getIMAPResponses();
+        final Long highestModSeq = infos.getHighestModSeq();
+
+        Assert.assertNull(highestModSeq, "Result mismatched.");
+        Assert.assertEquals(responses.size(), 0, "Result mismatched.");
+        Assert.assertEquals(modifiedMsgsets.length, 0, "Result mismatched.");
+    }
+
+    /**
+     * Tests StoreResult constructor and getters when passing highest mod sequence, empty fetch responses collection,
+     * and not passing modified message number collection.
+     */
+    @Test
+    public void testStoreResultIgnoreModifiedMessage() {
+        final StoreResult infos = new StoreResult(1L, Collections.emptyList());
+        final MessageNumberSet[] modifiedMsgsets = infos.getModifiedMsgSets();
+        final List<IMAPResponse>responses = infos.getIMAPResponses();
+        final long highestModSeq = infos.getHighestModSeq();
+
+        Assert.assertEquals(highestModSeq, 1L, "Result mismatched.");
+        Assert.assertEquals(responses.size(), 0, "Result mismatched.");
+        Assert.assertNull(modifiedMsgsets, "Result mismatched.");
+    }
+
+    /**
+     * Tests StoreResult constructor and getters when passing null highest mod sequence, empty fetch responses collection,
+     * and null modified message number collection.
+     */
+    @Test
+    public void testStoreResultNullModifiedMessage() {
```

Comment:
What is the difference between IgnoreModifiedMessage and NullModifiedMessage?

Reply:
Naming problem. It should be NullHighestModSeqNullModifiedMessage or OnlyImapResponses instead of NullModifiedMessage